### PR TITLE
New navigation

### PR DESCRIPTION
**Pull request by jonludlam**
Issue: #337 

Not quite ready to merge yet. Not quite sure what to do about attribution of the SVGs -- they're all from the noun project, which has requirements if you [download the icons for free](https://thenounproject.zendesk.com/hc/en-us/articles/200509928-How-do-I-give-creators-credit-in-my-work-). However, we may want to spend the $30-40 to get 'pro' versions that don't require attribution. 

The fonts don't look quite right to me either. I've switched to serif as per the design, and I'm sure there's a tailwind native way to sort out fonts site-wide, but I've not attempted that yet.

Additionally, the links aren't all going to the right place yet!

### CODE REVIEW

**Comment by jonludlam** [comment]
Screenshots:

Screenshot 2021-05-19 at 18 54 23
Screenshot 2021-05-19 at 18 54 09
Screenshot 2021-05-19 at 18 54 01
Screenshot 2021-05-19 at 18 53 42

**Comment by avsm** [comment]
Fixes https://github.com/ocaml/v3.ocaml.org/issues/337

File: src/GlobalData.res [hunk 1]
```diff
@@ -29,22 +31,84 @@ type navContent = {
...
+      label: `Language`,
+      url: InternalUrls.resourcesLanguage,
+      icon: Icons.language,
+      text: "Discover the OCaml tutorial, books and papers on OCaml, as well as the OCaml Manual",
+    },
+    packages: {
+      label: `Packages`,
+      url: InternalUrls.resourcesReleases,
+      icon: Icons.packages,
+      text: "Browse the many packages available in OCaml.",
```
Comment (avsm) [inline src/GlobalData.res:71]:
```suggestion
      text: "Browse the many packages published within the OCaml ecosystem.",
```

**Comment by agarwal** [comment]
@jonludlam Looks nice! I updated the PR description to indicate this is for issue #337.

File: src/NavEntry.res [hunk 2]
```diff
@@ -1,4 +1,6 @@
 type t = {
   label: string,
   url: string,
+  icon: React.element,
```
Comment (ghost) [inline src/NavEntry.res:4]: I was planning on refining the NavEntry data type. At present, the intention of the type is to hold data which is useful for both the header and the footer. Consider introducing a second type for data that is only useful for header navigation entries, if that sounds useful.

Comment (jonludlam) [reply src/NavEntry.res:4]: Sure, we can do that - let's get this in as-is though and fix the types in a later PR.

File: src/GlobalData.res [hunk 3]
```diff
@@ -29,22 +31,84 @@ type navContent = {
...
+    },
+    applications: {
+      label: `Applications`,
+      url: InternalUrls.resourcesApplications,
+      icon: Icons.applications,
+      text: "Using or building tools and applicatins in OCaml? This page is full of useful information.",
+    },
+    bestPractices: {
+      label: `Best Practices`,
+      url: InternalUrls.resourcesArchive,
```
Comment (ghost) [inline src/GlobalData.res]: As you noted in the PR description, please update this link.

File: src/GlobalData.res [hunk 4]
```diff
@@ -29,22 +31,84 @@ type navContent = {
...
-    archive: {label: `Archive`, url: InternalUrls.resourcesArchive},
+    language: {
+      label: `Language`,
+      url: InternalUrls.resourcesLanguage,
+      icon: Icons.language,
+      text: "Discover the OCaml tutorial, books and papers on OCaml, as well as the OCaml Manual",
+    },
+    packages: {
+      label: `Packages`,
+      url: InternalUrls.resourcesReleases,
```
Comment (ghost) [inline src/GlobalData.res]: Consider changing this to one of the staging sites for docs.ocaml.org.

File: src/GlobalData.res (hunk 3 above)
Comment (jonludlam) [reply src/GlobalData.res]: Ah, didn't spot that one, thanks!

File: src/GlobalData.res [hunk 5]
```diff
@@ -29,22 +31,84 @@ type navContent = {
...
+    },
+    industrialUsers: {
+      label: `Industrial Users`,
+      url: InternalUrls.industryUsers,
+      icon: Icons.industry,
+      text: "Discover the organisations and companies that use OCaml to accomplish their goals.",
+    },
+    academicExcellence: {
+      label: `Academic Excellence`,
+      url: InternalUrls.industrySuccessstories,
```
Comment (ghost) [inline src/GlobalData.res]: Please create a minimal Github issue to indicate that this page needs to be created. We can take care of creating it at a later point. Please add a TODO in the source code to note that this URL is intentionally wrong.

**Review by ghost — APPROVED** [review]
Feel free to merge at your discretion. Thanks again for getting involved!

I tried this deployed site at a few different screen sizes. I didn't try to understand the CSS modifications deeply, under the assumption that the CSS is a direct copy of the Tailwind UI sample.

File: src/GlobalData.res (hunk 4 above)
Comment (ghost) [reply src/GlobalData.res]: This was done in https://github.com/ocaml/v3.ocaml.org/pull/361/commits/8143becbac85e4e8f03933ce4e6d9c52e71cb782.

File: src/GlobalData.res (hunk 3 above)
Comment (ghost) [reply src/GlobalData.res]: This was done in https://github.com/ocaml/v3.ocaml.org/pull/361/commits/8143becbac85e4e8f03933ce4e6d9c52e71cb782.

File: src/GlobalData.res (hunk 5 above)
Comment (ghost) [reply src/GlobalData.res]: The issue is here: #370.

**Pull request merged**